src/routes/list/containers/BulkEditListItemsForm.jsx

```diff
@@ -0,0 +1,251 @@
+import axios from '../../../utils/api';
+import { itemName } from '../utils';
+import BulkEditListItemsFormFields from '../components/BulkEditListItemsFormFields';
+
+function BulkEditListItemsForm(props) {
+  // if no existing list options, new list form should be displayed by default
+  const existingListsOptions = props.lists.map((list) => ({ value: String(list.id), label: list.name }));
+
+  // attributes that makes sense to updated on all items are included
+  // set attributes to initial value if all items have the same value for the attribute
+  // the value of these attributes can be cleared for all items with their respective clear state attributes
+  // read, purchased, completed not included as they add complexity and they can be updated in bulk on the list itself
+  const initialAttr = (attribute) => {
+    const uniqValues = [...new Set(props.items.map(({ [attribute]: value }) => value))];
+    return uniqValues.length === 1 && uniqValues[0] ? uniqValues[0] : '';
+  };
+  const initialValues = {
+    copy: false,
+    move: false,
+    existingList: '',
+    newListName: '',
+    updateCurrentItems: false,
+    album: initialAttr('album'),
+    clearAlbum: false,
+    artist: initialAttr('artist'),
+    clearArtist: false,
+    assigneeId: String(initialAttr('assignee_id')),
+    clearAssignee: false,
+    author: initialAttr('author'),
+    clearAuthor: false,
+    category: initialAttr('category'),
+    clearCategory: false,
+    dueBy: formatDueBy(initialAttr('due_by')),
+    clearDueBy: false,
+    quantity: initialAttr('quantity'),
+    clearQuantity: false,
+    showNewListForm: !existingListsOptions.length,
+  };
+
+  const [formData, setFormData] = useState(initialValues);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const putData = {
+      category: formData.category || null,
+      author: formData.author || null,
+      quantity: formData.quantity || null,
+      artist: formData.artist || null,
+      album: formData.album || null,
+      assignee_id: formData.assigneeId || null,
+      due_by: formData.dueBy || null,
+      clear_category: formData.clearCategory,
+      clear_author: formData.clearAuthor,
+      clear_quantity: formData.clearQuantity,
+      clear_artist: formData.clearArtist,
+      clear_album: formData.clearAlbum,
+      clear_assignee: formData.clearAssignee,
+      clear_due_by: formData.clearDueBy,
+      // if copying, the user will set whether or not to update current items
+      // if moving, the current items will not be updated
+      // if not doing either, updating the current items is the only action being take
+      update_current_items: formData.copy ? formData.updateCurrentItems : !formData.move,
+    };
+    if (formData.copy) {
+      putData.copy = formData.copy;
+    }
+    if (formData.move) {
+      putData.move = formData.move;
+    }
+    if ((formData.copy || formData.move) && !formData.existingList && !formData.newListName) {
```

Comment:
may want to put this at the top of the function. why do any of this if the form data is invalid?